File: Ragdoll.java
import java.awt.*;

public abstract class Ragdoll {
    protected Sprite root;

    public enum Type {
        PERSON, TREE, DOG
    }

    public Sprite getSelectedSprite(Point p) {
        return this.root.getSelectedSprite(p);
    }

    public void draw(Graphics2D g) {
        this.root.draw(g);
    }
}

File: Ragdoll.kt
import java.awt.Graphics2D
import java.awt.Point

abstract class Ragdoll {
    abstract protected val root: Sprite

    enum class Type {
        PERSON, TREE, DOG
    }

    fun getSelectedSprite(p: Point): Sprite? {
        return root.getSelectedSprite(p)
    }

    fun draw(g: Graphics2D) {
        this.root.draw(g)
    }
}
